.github/workflows/cli_release_multiarch.yml

```diff
@@ -1,44 +1,44 @@
+---
 name: release cli multi architecture
-on:
+
+"on":
   release:
     branches:
       - 'go'
-    types: [released]
+    types:
+      - 'released'
+
 jobs:
   binary:
     strategy:
       matrix:
-        arch: [arm, arm64, amd64, 386]
+        arch: [arm, arm64, amd64, "386"]
         os: [linux, darwin, freebsd, windows]
         exclude:
           - os: darwin
             arch: arm
           - os: darwin
-            arch: 386
+            arch: "386"
 
     runs-on: ubuntu-latest
 
     steps:
-      - name: Download Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: 1.21.1
-        id: go
-
       - name: Check out repository
         uses: actions/checkout@v4
-      - name: Build
-        run: |
-          echo "Tag that is going to be used as digger version: ${{ github.event.release.tag_name }}"
-          env GOOS=${{matrix.os}} GOARCH=${{matrix.arch}} CGO_ENABLED=0 go build -ldflags="-X digger/pkg/utils.version=${{ github.event.release.tag_name }}" -o digger ./cli/cmd/digger
 
-      - name: Publish linux-x64 exec to github
-        id: upload-release-asset-linux-x64
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and publish binary artifact to GitHub
```

Comment:
I don't think there's much reason for concern when using third-party actions as long as we're pinning to the specific commit like we're doing so here. The code can't change from underneath you (as opposed to using tags that *could* get redefined). Unless licensing is your concern?

As for keeping this change out of the PR...that's kind of difficult since it technically is the bulk of this PR and iirc I couldn't figure out how to get and upload checksums in separate steps otherwise. But I'm more familiar with Actions now than I was before (using Digger actually is the first time I've used Actions) so maybe I could figure something out now. Not sure.

Then again my original incentive for this PR was that I wanted to add download verification to my Ansible role for deploying self-hosted Digger before publishing it but I've recently started using the helm release so it's kind of on the back-burner for me.

Reply:
OK, its mainly a security concern. The fact that its a hardcoded commit helps. I also would need to make not to not merge rennovate bumps unless the new version also reviewed. Lets merge it for now until we find a new solution